packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -111,7 +111,7 @@ export class Wallet {
       let isComplete = false;
 
       const onObjectiveSucceeded = (o: WalletObjective) => {
-        if (o.objectiveId === o.objectiveId) {
+        if (objective.objectiveId === o.objectiveId) {
```

Comment:
It looks like this flaw was copied from the original `ChannelManager` in the other repo.